ssl/statem/extensions.c

```diff
@@ -54,13 +52,14 @@ typedef struct extensions_definition_st {
      */
     int (*init)(SSL *s, unsigned int context);
     /* Parse extension sent from client to server */
-    int (*parse_ctos)(SSL *s, PACKET *pkt, int *al);
+    int (*parse_ctos)(SSL *s, PACKET *pkt, X509 *x, size_t chain, int *al);
```

Comment:
Ok, I changed the name to `chainidx`.

I *strongly* disagree on the type. The range of valid values exactly matches that of a size_t. It cannot be negative and the biggest (theoretical) value is SIZE_MAX - 1. The most common operations that you tend to perform on an index is to compare it to its boundary values...which is a size_t. Without that you have to perform casts to do the comparison. In fact there is scope for overflow errors if you use an int (INT_MAX != SIZE_MAX) and other oddities where the index goes -ve if its an int. A quick google search turned up a number of discussions on this issue (e.g. https://gustedt.wordpress.com/2013/07/15/a-praise-of-size_t-and-other-unsigned-types/).